dpppt-backend-sdk/dpppt-backend-sdk-ws/src/test/java/org/dpppt/backend/sdk/ws/insertmanager/InsertManagerTest.java

```diff
@@ -7,18 +7,75 @@
 import ch.qos.logback.classic.Logger;
 import ch.qos.logback.classic.spi.ILoggingEvent;
 import ch.qos.logback.core.AppenderBase;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
 import java.time.Duration;
+import java.time.Instant;
 import java.util.ArrayList;
+import java.util.Base64;
 import java.util.List;
+import java.util.UUID;
 import org.dpppt.backend.sdk.model.gaen.GaenKey;
 import org.dpppt.backend.sdk.semver.Version;
 import org.dpppt.backend.sdk.utils.UTCInstant;
 import org.dpppt.backend.sdk.ws.insertmanager.insertionmodifier.OldAndroid0RPModifier;
 import org.dpppt.backend.sdk.ws.util.ValidationUtils;
 import org.junit.Test;
 import org.slf4j.LoggerFactory;
+import org.springframework.security.oauth2.jwt.Jwt;
 
 public class InsertManagerTest {
+
+  protected String createToken(boolean fake, UTCInstant expiresAt) {
+    Claims claims = Jwts.claims();
+    claims.put("scope", "exposed");
+    claims.put("onset", "2020-04-20");
+    claims.put("fake", fake ? "1" : "0");
+    return Jwts.builder()
+        .setClaims(claims)
+        .setId(UUID.randomUUID().toString())
+        .setSubject("test-subject" + UTCInstant.now().getOffsetDateTime().toString())
+        .setExpiration(expiresAt.getDate())
+        .setIssuedAt(UTCInstant.now().getDate())
+        .compact();
+  }
+
+  @Test
+  public void fakeJWTDoesNotInsert() throws Exception {
+    InsertManager manager =
+        new InsertManager(
+            new MockDataSource(),
+            new ValidationUtils(16, Duration.ofDays(14), Duration.ofHours(2).toMillis()));
+    var jwt =
+        Jwt.withTokenValue(createToken(true, UTCInstant.now().plusMinutes(5)))
+            .header("alg", "HS256")
+            .claim("fake", "1")
+            .build();
+
+    var gaenKey1 = new GaenKey();
```

Comment:
This is hard to read and to understand why you have `gaenKey1` and `gaenKey2`.
Either rename them as `gaenKeyToday` and `gaenKeyYesterday`, or make an array.

Reply:
You are right, we probably should hide this from the actual test.
 
Just to note here, it  does not matter at all (other than the key should pass the filters) what is inserted, since we are bypassing any logic which is executed before (as e.g. checking for a valid JSON object and such)